lib/createUser.js

```diff
@@ -1,48 +1,35 @@
-
 const Sequelize = require('sequelize')
+const util = require('./util')
+const db = require('./db')
 
-var sha1 = require('sha1')
-
-var util = require('./util')
-
-var config = require('./config')
-
-var db = require('./db')
-
-function createUser (info) {
-  var graphUri =
-config.get('databasePrefix') + util.createTriplestoreID(info.username)
+const { User } = db.model
 
-  return db.model.User.findOrCreate({
-
-    where: Sequelize.or({
-      email: info.email
-    }, {
-      username: info.username
-    }),
+async function createUser (info) {
+  const graphUri = util.getUserGraphUri(info)
+  const [user, created] = await User.findOrCreate({
+    where: Sequelize.or(
+      { email: info.email },
+      { username: info.username }
+    ),
 
     defaults: {
       name: info.name,
       email: info.email,
       username: info.username,
       affiliation: info.affiliation,
-      password: sha1(config.get('passwordSalt') + sha1(info.password)),
-      graphUri: graphUri,
-      isAdmin: info.isAdmin !== undefined ? info.isAdmin : false,
-      isCurator: info.isCurator !== undefined ? info.isCurator : false,
-      isMember: info.isMember !== undefined ? info.isMember : false
+      password: User.hashPassword(info.password),
+      graphUri,
+      isAdmin: !!info.isAdmin,
```

Comment:
This is clever but a little difficult to parse if you are not familiar with how booleans should resolve. I think I prefer the old-style here.

Reply:
It's a common pattern used in JS applications. It guarantees that a variable will be converted to a boolean value.

I can revert the changes to the previous format if you want to.

Reply:
Ditto.  Please add a comment about the !! usage.  Took me a bit to sort out what you were trying to do.  Also, "graphUri," is a new one for me.  I'm assuming it assumes:  "graphUri: graphUri".  I think I prefer explicit here.

Reply:
@3ach @cjmyers I've made the code more explicit in 3eeeaf4

Reply:
Thanks